packages/comms/src/services/wsLogaccess.ts

```diff
@@ -1,8 +1,150 @@
-import { LogaccessServiceBase, WsLogaccess, LogAccessType } from "./wsdl/ws_logaccess/v1/ws_logaccess";
+import { scopedLogger } from "@hpcc-js/util";
+import { LogaccessServiceBase, WsLogaccess, LogAccessType, LogSelectColumnMode, LogAccessFilterOperator } from "./wsdl/ws_logaccess/v1.02/ws_logaccess";
+
+const logger = scopedLogger("@hpcc-js/comms/services/wsLogaccess.ts");
 
 export {
     WsLogaccess, LogAccessType
 };
 
+export enum KnownColumns {
+    audience = "hpcc.log.audience",
+    class = "hpcc.log.class",
+    jobId = "hpcc.log.jobid",
+    message = "hpcc.log.message",
+    procId = "hpcc.log.procid",
+    sequence = "hpcc.log.sequence",
+    threadId = "hpcc.log.threadid",
+    timestamp = "hpcc.log.timestamp",
+    containerName = "kubernetes.container.name"
+}
+
+const RKnownColumns: { [key: string]: string } = {};
+for (const key in KnownColumns) {
+    if (KnownColumns.hasOwnProperty(key)) {
+        RKnownColumns[KnownColumns[key]] = key;
+    }
+}
+
+export interface GetLogsExRequest {
+    audience?: string;
+    class?: string;
+    jobId?: string;
+    message?: string;
+    procId?: string;
+    sequence?: string;
+    threadId?: string;
+    timestamp?: string;
+    containerName?: string;
+    StartDate?: Date;
+    EndDate?: Date;
+    LogLineStartFrom: number,
+    LogLineLimit: number
+}
+
+export interface LogLine {
+    audience?: string;
+    class?: string;
+    jobId?: string;
+    message?: string;
+    procId?: number;
+    sequence?: string;
+    threadId?: number;
+    timestamp?: string;
+    containerName?: string;
+}
+
+const defaultToLogLine = (line?: any): LogLine => {
+    const retVal: LogLine = {};
+    for (const key in RKnownColumns) {
+        retVal[RKnownColumns[key]] = line?.fields[0][key];
+    }
+    return retVal;
+};
+
+export interface GetLogsExResponse {
+    lines: LogLine[],
+    total: number,
+}
+
 export class LogaccessService extends LogaccessServiceBase {
-}
\ No newline at end of file
+
+    GetLogAccessInfo(request: WsLogaccess.GetLogAccessInfoRequest): Promise<WsLogaccess.GetLogAccessInfoResponse> {
+        return super.GetLogAccessInfo(request);
+    }
+
+    GetLogs(request: WsLogaccess.GetLogsRequest): Promise<WsLogaccess.GetLogsResponse> {
+        return super.GetLogs(request);
+    }
+
+    GetLogsEx(request: GetLogsExRequest): Promise<GetLogsExResponse> {
+        const getLogsRequest: WsLogaccess.GetLogsRequest = {
+            Filter: {
+                leftBinaryFilter: {
+                    BinaryLogFilter: [{
+                        leftFilter: {
+                            LogCategory: LogAccessType.All,
```

Comment:
Should this be checking for the presence of a category in the request and defaulting to the ALL if not found?